diff --git a/src/drf/SmallStep.v b/src/drf/SmallStep.v
index 7884c26..bc84372 100644
--- a/src/drf/SmallStep.v
+++ b/src/drf/SmallStep.v
@@ -277,6 +277,11 @@ Proof.
   rewrite (IdentMap.add_add_eq tid (existT _ lang st2, lc2)). eauto.
 Qed.
 
+Arguments Thread.state {lang}.
+Arguments Thread.local {lang}.
+Arguments Thread.sc {lang}.
+Arguments Thread.memory {lang}.
+
 Lemma rtc_thread_step_rtc_small_step_aux
       lang tid threads
       th1 th2
