diff --git a/src/opt/MergeStep.v b/src/opt/MergeStep.v
index 621a8fe..3ae2025 100644
--- a/src/opt/MergeStep.v
+++ b/src/opt/MergeStep.v
@@ -84,6 +84,8 @@ Proof.
     etrans; apply WF0.
 Qed.
 
+Notation "x .( y )" := (y x) (at level 0, only parsing).
+
 Lemma merge_write_read2
       loc from to val releasedm released ord1 ord2 kind
       lc0 sc0 mem0
diff --git a/src/opt/SplitAcq.v b/src/opt/SplitAcq.v
index 5c161f3..cc4b5e7 100644
--- a/src/opt/SplitAcq.v
+++ b/src/opt/SplitAcq.v
@@ -65,6 +65,8 @@ Proof.
   - econs; eauto.
 Qed.
 
+Notation "x .( y )" := (y x) (at level 0, only parsing).
+
 Lemma sim_local_fulfill_acquired
       lc1_src sc1_src mem1_src
       lc1_tgt sc1_tgt mem1_tgt
